### ISSUE TITLE
Global Random found Harmful

### ISSUE THREAD

**Issue by PDoakORNL** [QMCPACK/qmcpack#1983]
So 
```C++
NonLocalECPotential::NonLocalECPotential(ParticleSet& ions,
                                         ParticleSet& els,
                                         TrialWaveFunction& psi,
                                         bool computeForces,
                                         bool useVP)
    : IonConfig(ions),
      Psi(psi),
      UseTMove(TMOVE_OFF),
      myRNG(&Random),
```
Random is a global object and not threadsafe.  In the new DMCBatched driver NonLocalECPPotential::evaluate is called in a threaded context.  It does this
```C++
   PPset[ipp]->randomize_grid(*myRNG);
```
This rather randomly causes SIGSEGV

This took me quite some time to catch in a core dump. Enough with the global objects and unclear scopes.

**Comment by jtkrogel** [comment]
What are the implications for current use of random numbers for different threads?

**Comment by ye-luo** [comment]
The global RNG recorded in the constructor is only used by the CUDA code.
The CPU code path overwrites the recorded one by the per-thread RNG at a later stage by calling setRandomGenerator. So there is no problem in the production code.

**Comment by prckent** [comment]
...and the CUDA code does not do threaded execution, so production use is safe even if ugly.

**Comment by PDoakORNL** [comment]
I still have a segfault. CoreDump points here.

**Comment by PDoakORNL** [comment]
This is only surfaced for the DMCBatched, I not implying current code isn't a special case where this somehow works.  But I'm not participating in further development impacted by this until I fix it and I don't much care about the state manipulation that makes it work in the legacy code.

**Comment by ye-luo** [comment]
Yes. This is a valid bug report. We probably need to find a clean way to deal with per walker RNG need.
